File: clean-json.py
#!/usr/bin/env python3
import json

# Read the skills.json file
with open('data/skills.json', 'r') as f:
    skills = json.load(f)

# Clean each skill
cleaned_skills = []
for skill in skills:
    cleaned = {
        "name": skill["name"],
        "requirements": skill["requirements"],
        "wiki_url": skill["wiki_url"]
    }

    # Clean ability_details
    details = {
        "ap_cost": skill["ability_details"]["ap_cost"],
        "sp_cost": skill["ability_details"]["sp_cost"],
        "range": skill["ability_details"]["range"],
        "cooldown": skill["ability_details"]["cooldown"],
        "effect": skill["ability_details"]["effect"]
    }

    # Only include special_terms if it has values
    if skill["ability_details"].get("special_terms") and len(skill["ability_details"]["special_terms"]) > 0:
        details["special_terms"] = skill["ability_details"]["special_terms"]

    cleaned["ability_details"] = details
    cleaned_skills.append(cleaned)

# Write back to file with proper formatting
with open('data/skills.json', 'w') as f:
    json.dump(cleaned_skills, f, indent=2)
    f.write('\n')

print('✓ Cleaned skills.json')
print('  - Removed: source_cost, description, found_on_page, has_wiki_page')
print('  - Removed: empty special_terms arrays')
print(f'  - Processed {len(cleaned_skills)} skills')

File: clean-json.js
const fs = require('fs');

// Read the skills.json file
const skills = JSON.parse(fs.readFileSync('data/skills.json', 'utf8'));

// Clean each skill
const cleanedSkills = skills.map(skill => {
  const cleaned = {
    name: skill.name,
    requirements: skill.requirements,
    has_wiki_page: skill.has_wiki_page,
    wiki_url: skill.wiki_url
  };

  // Clean ability_details
  const details = {};
  details.ap_cost = skill.ability_details.ap_cost;
  details.sp_cost = skill.ability_details.sp_cost;
  details.range = skill.ability_details.range;
  details.cooldown = skill.ability_details.cooldown;
  details.effect = skill.ability_details.effect;

  // Only include special_terms if it has values
  if (skill.ability_details.special_terms && skill.ability_details.special_terms.length > 0) {
    details.special_terms = skill.ability_details.special_terms;
  }

  cleaned.ability_details = details;

  return cleaned;
});

// Write back to file with proper formatting
fs.writeFileSync('data/skills.json', JSON.stringify(cleanedSkills, null, 2) + '\n');

console.log('✓ Cleaned skills.json');
console.log('  - Removed: source_cost, description, found_on_page');
console.log('  - Removed: empty special_terms arrays');
console.log(`  - Processed ${cleanedSkills.length} skills`);
